src/python/grpcio/grpc/experimental/aio/_channel.py

```diff
@@ -322,6 +366,53 @@ def __init__(self, target: Text, options: Optional[ChannelArgumentType],
         self._loop = asyncio.get_event_loop()
         self._channel = cygrpc.AioChannel(_common.encode(target), options,
                                           credentials, self._loop)
+        self._ongoing_calls = _OngoingCalls()
+
+    async def __aenter__(self):
+        """Starts an asynchronous context manager.
+
+        Returns:
+          Channel the channel that was instantiated.
+        """
+        return self
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        """Finishes the asynchronous context manager by closing gracefully the channel."""
+        await self._close()
+
+    async def _wait_for_close_ongoing_calls(self):
+        sleep_iterations_sec = 0.001
+
+        while self._ongoing_calls.size() > 0:
+            await asyncio.sleep(sleep_iterations_sec)
```

Comment:
Busy loop is not performing... Maybe we need a mechanism like Golang's wait-group.

Reply:
You made the point, technically an `asyncio.gather(*calls, timeout)` should work (the equivalent of Golang's wait-group). Thanks, I will try it.